core/src/lib.rs

```diff
@@ -329,6 +329,91 @@ impl From<f64> for TypedValue {
     }
 }
 
+pub trait TypedValueConverter {
+    fn as_entid(self) -> Option<Entid>;
+    fn as_kw(self) -> Option<Rc<NamespacedKeyword>>;
```

Comment:
… but if this is for the FFI, you probably don't want to be leaking `Rc`, and perhaps not even `NamespacedKeyword`…